bzerolib/report/errorreport.go

```
@@ -14,33 +18,35 @@ const (
 type ErrorReport struct {
 	Reporter  string      `json:"reporter"`
 	Timestamp string      `json:"timestamp"`
-	State     interface{} `json:"state"`
 	Message   string      `json:"message"`
+	State     interface{} `json:"state"`
 	Logs      string      `json:"logs"`
 }
 
-func ReportError(logger *logger.Logger, serviceUrl string, errReport ErrorReport) {
-	// make our state a string
-	stateBytes, err := json.Marshal(errReport.State)
-	if err != nil {
-		logger.Errorf("error marshalling error report: %+v", errReport)
-		return
-	}
-	errReport.State = string(stateBytes)
-
-	endpoint, err := bzhttp.BuildEndpoint(serviceUrl, errorEndpoint)
-	if err != nil {
-		logger.Errorf("failed to build error report endpoint: %+v", err)
-	}
+func ReportError(logger *logger.Logger, ctx context.Context, serviceUrl string, errReport ErrorReport) {
```

Wouldn't it make more sense for this function to just return an error instead of taking a logger as an argument? I know this predates your PR but while you're in here I think it would be the right thing to do.